packages/flutter/lib/src/cupertino/date_picker.dart

```diff
@@ -1440,16 +1453,23 @@ class _CupertinoDatePickerDateState extends State<CupertinoDatePicker> {
       ));
     }
 
+    final double maxPickerWidth = totalColumnWidths > _kPickerWidth ? totalColumnWidths : _kPickerWidth;
+
     return MediaQuery(
       data: MediaQuery.of(context).copyWith(textScaleFactor: 1.0),
       child: DefaultTextStyle.merge(
         style: _kDefaultPickerTextStyle,
-        child: CustomMultiChildLayout(
-          delegate: _DatePickerLayoutDelegate(
-            columnWidths: columnWidths,
-            textDirectionFactor: textDirectionFactor,
+        child: Center(
+          child: Container(
```

Comment:
Container has an `alignment` args in its constructor.

Reply:
Does the other picker need the same treatmeant?

Reply:
Maybe move the logic to `_DatePickerLayoutDelegate` so you don't have to copy paste it.

Reply:
I moved the majority of the logic to `_DatePickerLayoutDelegate`. 👍